src/fabric_cli/utils/fab_cmd_set_utils.py

```diff
@@ -6,9 +6,32 @@
 
 from fabric_cli.core import fab_constant, fab_logger
 from fabric_cli.core.fab_exceptions import FabricCLIError
+from fabric_cli.errors.common import CommonErrors
 from fabric_cli.utils import fab_jmespath as utils_jmespath
 
 
+def validate_item_query(query_value: str) -> None:
+    """Validate that a query string is allowed for item metadata modification.
+
+    Args:
+        query_value: Query string to validate. Must be an allowed metadata key,
+            or start with "definition." or "properties.".
+
+    Raises:
+        FabricCLIError: If the query is not in an allowed format.
+    """
+    if not (
+        query_value in fab_constant.ITEM_SET_ALLOWED_METADATA_KEYS
+        or query_value == fab_constant.ITEM_QUERY_DEFINITION
+        or query_value.startswith(f"{fab_constant.ITEM_QUERY_DEFINITION}.")
+        or query_value.startswith(f"{fab_constant.ITEM_QUERY_PROPERTIES}.")
```

Comment:
what about a scenario of querying the entire `properties` object? 

like

fab set <item> -q properties -i {prop1: value, prop2: value2}